### ISSUE TITLE
Move x3d base creation to reusable helper

### ISSUE THREAD

**Issue by jamesleesaunders** [jamesleesaunders/d3-x3d#173]
At the moment every chart component has the same identical lines for the initial creation of the x3d, scene and viewpoint tags. This should be broken out into a single reusable component to make code DRYer!

Code like:
```
	/**
	 * Create X3D base and scene
	 *
	 * @param selection
	 * @param layers
	 */
	function createBase(selection, layers) {
		// Create x3d element (if it does not exist already)
		if (!x3d) {
			x3d = selection.append("X3D");
			scene = x3d.append("Scene");

			x3d.attr("width", width + "px")
				.attr("height", height + "px")
				.attr("showLog", debug ? "true" : "false")
				.attr("showStat", debug ? "true" : "false")
				.attr("useGeoCache", false);

			// Disable gamma correction
			scene.append("Environment")
				.attr("gammaCorrectionDefault", "none");

			// Add a white background
			scene.append("Background")
				.attr("groundColor", "1 1 1")
				.attr("skyColor", "1 1 1");
		}

		// Add layer groups
		scene.classed(classed, true)
			.selectAll("Group")
			.data(layers)
			.enter()
			.append("Group")
			.attr("class", (d) => d);
	}
```
This same code can be found in every one of the /src/chart/ files.

**Comment by mo-jinbuu** [comment]
Hey @jamesleesaunders is this still available? I'd like to work on it

**Comment by jamesleesaunders** [comment]
Hi @mo-jinbuu sure! yes it still needs doing. I would be very excited for you to look into this.
Let me know if you have any more questions or ideas.

**Comment by mo-jinbuu** [comment]
#180 Created a PR. Let me know if I need to fix anything. I ran the test script ('npm run test') on my local machine and had everything passing. Here's the log: 
[test_results.txt](https://github.com/jamesleesaunders/d3-x3d/files/5350926/test_results.txt)

**Comment by jamesleesaunders** [comment]
Closed by #180 Thanks @mo-jinbuu !
